Added test_rancher_cli.py, modified common_fixtures.py and related fi…

soumyalj opened the pull request:
…les.
@sangeethah : Checked in rancher cli test cases and related files. All test cases have passed.
Please take a look.

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:
```diff
@@ -2903,3 +2904,116 @@ def create_service_ingress(ingresses, services, port, namespace,
+
+def execute_rancher_cli(client, env_name, command,
+                        expected_resp, docker_compose, rancher_compose,
+                        timeout=SERVICE_WAIT_TIMEOUT):
```
expected_Rep should be removed since it is not used
Have default values as None for docker_compose and rancher_compose
Rename env_name as stack_name

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:
```diff
@@ -2903,3 +2904,116 @@ def create_service_ingress(ingresses, services, port, namespace,
+               " -f " + docker_filename
+    else:
+        cmd6 = "./rancher " + command
+
```
can we reduce the elseif loop here?

soumyalj replied:
If else loop has been condensed.

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/test_rancher_cli.py:
```diff
@@ -0,0 +1,620 @@
+    cli_response = execute_rancher_cli(client, stack_name, command,
+                                       service.id, docker_compose=None,
+                                       rancher_compose=None)
+    service = client.wait_success(service, 300)
```
Do we need this ? Should cli return only after the command succeeds ?

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/test_rancher_cli.py:
```diff
@@ -0,0 +1,620 @@
+    container_list = get_service_container_list(admin_client, service)
+    assert len(container_list) == 2
+    for container in container_list:
+        assert container.state == "running"
```
check for startCount to be 1

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/test_rancher_cli.py:
```diff
@@ -0,0 +1,620 @@
+                                        networkMode=MANAGED_NETWORK,
+                                        imageUuid=TEST_IMAGE_UUID)
+
+    container = client.wait_success(container, 60)
```
Include tests for stop/start/restart for containers that are part of service

soumyalj replied:
Included test for restarting containers that are part of a service. Test for stopping containers that  are part of a service will be done in the next checkin.

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/test_rancher_cli.py:
```diff
@@ -0,0 +1,620 @@
+    assert service2.scale == 2
+    assert service2.name == "rtest8-two"
+
+    command = "ps -a"
```
make sure that you have stopped services that get listed

soumyalj replied:
Performed stop of a service and made sure that stopped services are listed too.

sangeethah commented on tests/v2_validation/cattlevalidationtest/core/test_rancher_cli.py:
```diff
@@ -0,0 +1,620 @@
+    stack_name = random_str().replace("-", "")
+
+    command = "env ls"
+    expected_response = "cattle"
```
Expected response should be able to test for name and orchestration

soumyalj replied:
cattle and Default are included in the expected response.

soumyalj commented:
@sangeethah : tests for start/stop containers that are part of a service will be implemented in my next PR. I have added a test for restart container which is part of a service and addressed the comments.
